File: wpilibNewCommands/src/test/native/cpp/command/EndlessCommandTest.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "CommandTestBase.h"
#include "frc/command/EndlessCommand.h"
#include "frc/command/InstantCommand.h"

using namespace frc;
class EndlessCommandTest : public CommandTestBase {};

TEST_F(EndlessCommandTest, EndlessCommandSchedule) {
  CommandScheduler scheduler = GetScheduler();

  bool check = false;

  EndlessCommand command{InstantCommand([&check] { check = true; }, {})};

  scheduler.Schedule(&command);
  scheduler.Run();
  EXPECT_TRUE(scheduler.IsScheduled(&command));
  EXPECT_TRUE(check);
}

File: wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj/command/EndlessCommandTest.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package edu.wpi.first.wpilibj.command;

import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class EndlessCommandTest extends CommandTestBase {
  @Test
  void endlessCommandScheduleTest() {
    try (CommandScheduler scheduler = new CommandScheduler()) {
      EndlessCommand command = new EndlessCommand(new InstantCommand());

      scheduler.schedule(command);
      scheduler.run();

      assertTrue(scheduler.isScheduled(command));
    }
  }
}
